### ISSUE TITLE
Symbol's value as variable is void: hybrid-mode-use-evil-search-module

### ISSUE THREAD

**Issue by swaroopch** [syl20bnr/spacemacs#8321]
#### Description :octocat:
When pulling `develop` branch with https://github.com/syl20bnr/spacemacs/commit/c7e47de9383f7556553c531ff27e7654b3cba3db , Spacemacs refuses to start, there's no evil mode and even `M-m` doesn't work.

#### Reproduction guide :beetle:
- git checkout develop
- git pull origin
- Start Emacs

*Observed behaviour:* :eyes: :broken_heart:

```
Warning (initialization): An error occurred while loading ‘/Users/swaroop/.emacs.d/init.el’:

Symbol's value as variable is void: hybrid-mode-use-evil-search-module

To ensure normal operation, you should investigate and remove the
cause of the error in your initialization file.  Start Emacs with
the ‘--debug-init’ option to view a complete error backtrace.
```

*Expected behaviour:* :heart: :smile:

Should start normally

#### Backtrace :paw_prints:
```
Debugger entered--Lisp error: (void-variable hybrid-mode-use-evil-search-module)
  (and (eq (quote hybrid) style) hybrid-mode-use-evil-search-module)
  (or (eq (quote vim) style) (and (eq (quote hybrid) style) hybrid-mode-use-evil-search-module))
  (cond ((or (eq (quote vim) style) (and (eq (quote hybrid) style) hybrid-mode-use-evil-search-module)) (setq-default evil-search-module (quote evil-search))) (t (setq-default evil-search-module (quote isearch))))
  spacemacs/set-evil-search-module(hybrid)
  spacemacs-bootstrap/init-evil()
  funcall(spacemacs-bootstrap/init-evil)
  (let* ((pkg-name (eieio-oref pkg (quote :name))) (owner (car (eieio-oref pkg (quote :owners))))) (spacemacs-buffer/message (format "Configuring %S..." pkg-name)) (mapc (function (lambda (layer) (if (configuration-layer/layer-usedp layer) (progn (if (not ...) (spacemacs-buffer/message ...) (spacemacs-buffer/message ...) (condition-case err ... ...)))))) (eieio-oref pkg (quote :pre-layers))) (spacemacs-buffer/message (format "  -> init (%S)..." owner)) (funcall (intern (format "%S/init-%S" owner pkg-name))) (mapc (function (lambda (layer) (if (configuration-layer/layer-usedp layer) (progn (if (not ...) (spacemacs-buffer/message ...) (spacemacs-buffer/message ...) (condition-case err ... ...)))))) (eieio-oref pkg (quote :post-layers))))
  configuration-layer//configure-package([eieio-class-tag--cfgl-package evil nil (spacemacs-bootstrap) nil (emacs-lisp) elpa t bootstrap nil t nil])
  (cond ((eq (quote dotfile) (car (eieio-oref pkg (quote :owners)))) (spacemacs-buffer/message (format "%S is configured in the dotfile." pkg-name))) (t (configuration-layer//configure-package pkg)))
  (cond ((eieio-oref pkg (quote :lazy-install)) (spacemacs-buffer/message (format "%S ignored since it can be lazily installed." pkg-name))) ((and (eieio-oref pkg (quote :excluded)) (not (eieio-oref pkg (quote :protected)))) (spacemacs-buffer/message (format "%S ignored since it has been excluded." pkg-name))) ((null (eieio-oref pkg (quote :owners))) (spacemacs-buffer/message (format "%S ignored since it has no owner layer." pkg-name))) ((not (cfgl-package-enabledp pkg)) (spacemacs-buffer/message (format "%S is toggled off." pkg-name))) (t (let ((dir (configuration-layer/get-location-directory pkg-name (eieio-oref pkg (quote :location)) (car (eieio-oref pkg ...))))) (if dir (progn (add-to-list (quote load-path) dir)))) (if (memq (eieio-oref pkg (quote :location)) (quote (local site built-in))) nil (configuration-layer//activate-package pkg-name)) (cond ((eq (quote dotfile) (car (eieio-oref pkg (quote :owners)))) (spacemacs-buffer/message (format "%S is configured in the dotfile." pkg-name))) (t (configuration-layer//configure-package pkg)))))
  (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg (quote :lazy-install)) (spacemacs-buffer/message (format "%S ignored since it can be lazily installed." pkg-name))) ((and (eieio-oref pkg (quote :excluded)) (not (eieio-oref pkg (quote :protected)))) (spacemacs-buffer/message (format "%S ignored since it has been excluded." pkg-name))) ((null (eieio-oref pkg (quote :owners))) (spacemacs-buffer/message (format "%S ignored since it has no owner layer." pkg-name))) ((not (cfgl-package-enabledp pkg)) (spacemacs-buffer/message (format "%S is toggled off." pkg-name))) (t (let ((dir (configuration-layer/get-location-directory pkg-name (eieio-oref pkg ...) (car ...)))) (if dir (progn (add-to-list (quote load-path) dir)))) (if (memq (eieio-oref pkg (quote :location)) (quote (local site built-in))) nil (configuration-layer//activate-package pkg-name)) (cond ((eq (quote dotfile) (car (eieio-oref pkg ...))) (spacemacs-buffer/message (format "%S is configured in the dotfile." pkg-name))) (t (configuration-layer//configure-package pkg))))))
  (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (spacemacs-buffer/loading-animation) (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg (quote :lazy-install)) (spacemacs-buffer/message (format "%S ignored since it can be lazily installed." pkg-name))) ((and (eieio-oref pkg (quote :excluded)) (not (eieio-oref pkg (quote :protected)))) (spacemacs-buffer/message (format "%S ignored since it has been excluded." pkg-name))) ((null (eieio-oref pkg (quote :owners))) (spacemacs-buffer/message (format "%S ignored since it has no owner layer." pkg-name))) ((not (cfgl-package-enabledp pkg)) (spacemacs-buffer/message (format "%S is toggled off." pkg-name))) (t (let ((dir (configuration-layer/get-location-directory pkg-name ... ...))) (if dir (progn (add-to-list ... dir)))) (if (memq (eieio-oref pkg (quote :location)) (quote (local site built-in))) nil (configuration-layer//activate-package pkg-name)) (cond ((eq (quote dotfile) (car ...)) (spacemacs-buffer/message (format "%S is configured in the dotfile." pkg-name))) (t (configuration-layer//configure-package pkg)))))) (setq --dolist-tail-- (cdr --dolist-tail--)))
  (let ((--dolist-tail-- packages) pkg-name) (while --dolist-tail-- (setq pkg-name (car --dolist-tail--)) (spacemacs-buffer/loading-animation) (let ((pkg (configuration-layer/get-package pkg-name))) (cond ((eieio-oref pkg (quote :lazy-install)) (spacemacs-buffer/message (format "%S ignored since it can be lazily installed." pkg-name))) ((and (eieio-oref pkg (quote :excluded)) (not (eieio-oref pkg ...))) (spacemacs-buffer/message (format "%S ignored since it has been excluded." pkg-name))) ((null (eieio-oref pkg (quote :owners))) (spacemacs-buffer/message (format "%S ignored since it has no owner layer." pkg-name))) ((not (cfgl-package-enabledp pkg)) (spacemacs-buffer/message (format "%S is toggled off." pkg-name))) (t (let ((dir ...)) (if dir (progn ...))) (if (memq (eieio-oref pkg ...) (quote ...)) nil (configuration-layer//activate-package pkg-name)) (cond ((eq ... ...) (spacemacs-buffer/message ...)) (t (configuration-layer//configure-package pkg)))))) (setq --dolist-tail-- (cdr --dolist-tail--))))
  configuration-layer//configure-packages-2((async bind-key bind-map diminish evil hydra use-package which-key))
  configuration-layer//configure-packages((abbrev ac-ispell ace-jump-helm-line ace-link ace-window adaptive-wrap afternoon-theme aggressive-indent alect-themes ample-theme ample-zen-theme anaconda-mode ansi-colors anti-zenburn-theme apropospriate-theme archive-mode async auto-compile auto-complete auto-highlight-symbol auto-yasnippet avy badwolf-theme bind-key bind-map birds-of-paradise-plus-theme bookmark bracketed-paste bubbleberry-theme bundler busybee-theme cc-mode centered-buffer-mode centered-cursor cherry-blossom-theme chruby clang-format clean-aindent-mode clues-theme cmake-mode coffee-mode color-theme-sanityinc-solarized color-theme-sanityinc-tomorrow column-enforce-mode comint company company-anaconda company-c-headers company-go company-quickhelp ...))
  configuration-layer/sync()
  (if (not (version<= spacemacs-emacs-min-version emacs-version)) (message (concat "Your version of Emacs (%s) is too old. " "Spacemacs requires Emacs version %s or above.") emacs-version spacemacs-emacs-min-version) (load-file (concat (file-name-directory load-file-name) "core/core-load-paths.el")) (require (quote core-spacemacs)) (spacemacs/init) (configuration-layer/sync) (spacemacs-buffer/display-startup-note) (spacemacs/setup-startup-hook) (require (quote server)) (if (server-running-p) nil (server-start)))
  eval-buffer(#<buffer  *load*> nil "/Users/swaroop/.emacs.d/init.el" nil t)  ; Reading at buffer position 1270
  load-with-code-conversion("/Users/swaroop/.emacs.d/init.el" "/Users/swaroop/.emacs.d/init.el" t t)
  load("/Users/swaroop/.emacs.d/init" t t)
  #[0 "\205\266	\306=\203\307\310Q\202?	\311=\204\307\312Q\202?\313\307\314\315#\203*\316\202?\313\307\314\317#\203>\320\321\322!D\nB\323\202?\316\324\325\324\211#\210\324=\203e\326\327\330\307\331Q!\"\325\324\211#\210\324=\203d\210\203\247\332!\333\232\203\247\334!\211\335P\336!\203\201\211\202\214\336!\203\213\202\214\314\262\203\245\337\"\203\243\340\341#\210\342\343!\210\266\f?\205\264\314\325\344\324\211#)\262\207" [init-file-user system-type delayed-warnings-list user-init-file inhibit-default-init inhibit-startup-screen ms-dos "~" "/_emacs" windows-nt "/.emacs" directory-files nil "^\\.emacs\\(\\.elc?\\)?$" "~/.emacs" "^_emacs\\(\\.elc?\\)?$" initialization format-message "`_emacs' init file is deprecated, please use `.emacs'" "~/_emacs" t load expand-file-name "init" file-name-as-directory "/.emacs.d" file-name-extension "elc" file-name-sans-extension ".el" file-exists-p file-newer-than-file-p message "Warning: %s is newer than %s" sit-for 1 "default"] 7]()
  command-line()
  normal-top-level()
```

**Comment by swaroopch** [comment]
Note that `git checkout d3bd29997a44ff7689906d89637c71d77ae5d3f3` (before the c7e47de9383f7556553c531ff27e7654b3cba3db commit) doesn't have this issue and Spacemacs functions normally.

**Comment by deb0ch** [comment]
A quick fix is available here: #8317.

Not sure that it should be the definitive fix, but at least it fixed the issue for me.

**Comment by swaroopch** [comment]
@deb0ch That patch works for me.

**Comment by nixmaniack** [comment]
Another duplicate after #8304, #8305.

**Comment by swaroopch** [comment]
Closing as duplicate of #8304

**Comment by syl20bnr** [comment]
Fixed in develop.
